src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java

```diff
@@ -474,6 +477,75 @@ public StudyDetailsDto getStudyDetailsByInstance(final Integer instanceId) {
 		}
 	}
 
+	@Override
+	public List<StudyInstanceDto> getStudyInstanceDtoListWithTrialData(final StudySearchFilter studySearchFilter, final Pageable pageable) {
+		try {
+			final List<StudyInstanceDto> studyInstanceDtos = this.daoFactory.getDmsProjectDAO()
+				.getStudyInstances(studySearchFilter, pageable);
+			final Map<Integer, List<ObservationLevel>> trialIdToObservationLevelMap = new HashMap<>();
+			if(studyInstanceDtos != null) {
+				for(final StudyInstanceDto studyInstanceDto: studyInstanceDtos) {
+					final Integer trialDbId = Integer.valueOf(studyInstanceDto.getTrialDbId());
+					final Integer studyDbId = Integer.valueOf(studyInstanceDto.getStudyDbId());
+					final DmsProject environmentDataset =
```

Comment:
I am concerned about the performance of retrieving environment details and conditions inside the loop.
I think we should keep maps so that it will be retrieved only once for all environments of a study:
`Map<Integer, List<MeasurementVariable>> studyEnvironmentConditionsMap`
`Map<Integer, List<MeasurementVariable>> studyEnvironmentDetailsMap`